### DISCUSSION TITLE
Troubles with wayvnc

### DISCUSSION THREAD

**Discussion by rytec-nl** [hyprwm/Hyprland#974] [Q&A]
Hello,

I setup my systems with Hyprland and really like it so far. Seeing it is build upon wlroots I wanted to use wayvnc to login to my systems remotely. However, using wayvnc seems to cause some weird behavior with inputs:

- The  mouse cursor behaves as it is frozen in the location it was when wayvnc is started, despite `hyprctl cursorpos` updating normally. The mouse pointer icon also does not update. The cursor is drawn in the correct location, its that left or right mouse button input is in the old frozen position.

- Keyboard input on the local system remains functional but only "normal" key inputs work through VNC. Using modifier keys does not seem to work, which makes using the desktop quit difficult :). I configured a submap for passthrough, but that does not help unfortunately

At this point in time I don't know where to look for a cause of the issues, whether it is configuration for wayvnc, Hyprland or something else on my systems is interfering (two seperate system behave the same but are based on the same NixOs config).

Starting wayvnc shows the following in Hyprland logs:
```
LOG] Window 2901b30 set title to wayvnc ~
[LOG] Registered signal for owner 22beb50: 2877fc8 -> 22beb58 (owner: Keyboard)
[LOG] Registered signal for owner 22beb50: 2877fb8 -> 22bebc0 (owner: Keyboard)
[LOG] Registered signal for owner 22beb50: 2877e98 -> 22bec90 (owner: Keyboard)
[LOG] Registered signal for owner 22beb50: 2877fd8 -> 22bec28 (owner: Keyboard)
[LOG] ApplyConfigToKeyboard for "wlr_virtual_keyboard_v1", hasconfig: 0
[LOG] Attempting to create a keymap for layout us with variant euro (rules: , model: , options: )
[LOG] Set the keyboard layout to us and variant to euro for keyboard "wlr_virtual_keyboard_v1"
[LOG] New virtual keyboard created, pointers Hypr: 22beb50 and WLR: 2877e80
[LOG] Applied config to mouse MOSART Semi. 2.4G Keyboard Mouse, sens 0.00
[LOG] Applied config to mouse MOSART Semi. 2.4G Keyboard Mouse Consumer Control, sens 0.00
[LOG] Applied config to mouse ELAN0001:00 04F3:3140 Mouse, sens 0.00
[LOG] Applied config to mouse ELAN0001:00 04F3:3140 Touchpad, sens 0.00
[LOG] Registered signal for owner 28b5a50: 28b1858 -> 28b5b10 (owner: Mouse)
[LOG] New mouse created, pointer WLR: 28b1840
```

I start wayvnc with the following config (setting --log-level=trace has surprisingly little output):
```
address=0.0.0.0
xkb_layout=us
xkb_variant=euro
xkb_options=compose:menu
```
I connected using `realvnc-vnc-viewer` and `remmina`.

Could you perhaps point me in the right direction to start debugging?  Please let me know if additional information is required.

Rens

**Comment by vaxerski** [comment]
Can you post a mini-tutorial so I can test this

**Comment by rytec-nl** [comment]
Of course:

- Install wayvnc. Using Nix/NixOs: `nix shell nixpkgs#wayvnc` or look here for another distro: https://github.com/any1/wayvnc#installing
- From within Hyprland run `wayvnc 0.0.0.0`

For me the immediately freezes the mouse input to the last cursor location.

Then for keyboard input, connect from another system using a VNC client such as remmina. I am unable to pass any modifier key to trigger a bind.

Feel free to ask for further clarification.

**Reply by vaxerski** [reply]
![image](https://user-images.githubusercontent.com/43317083/200422765-ab5684aa-bd5f-4688-83fe-84431b1c05ba.png)
let's do this

**Reply by vaxerski** [reply]
good news and good news

I reproduced the issue

**Reply by rytec-nl** [reply]
That is very good news, because that means I'm not an (complete) idiot. Let me know if I can do anything to help.

**Reply by vaxerski** [reply]
all fixed in aefc34b4050ec2d4aa70220726c168ca6ad5eea2

![image](https://user-images.githubusercontent.com/43317083/200428198-c6fb244a-92c9-4e55-844d-6907017b2c03.png)

**Reply by rytec-nl** [reply]
Damn, you're fast! I will check it out as soon as I can, which is probably tomorrow.

**Reply by vaxerski** [reply]
it's because I have no life or friends, and I have never seen a woman.

**Reply by rytec-nl** [reply]
That would make you fast in more ways then one ;) But you satisfied my needs, it works perfectly! Thanks!
